### PR TITLE
add password and hidden parameter types

### PR DESCRIPTION
**Pull request by loomis**
Password and hidden parameter types are needed to help the clients decide which parameters to make visible to users.  

Connected to SixSq/tasklist#1138.

### CODE REVIEW

File: jar-persistence/src/main/java/com/sixsq/slipstream/persistence/ParameterType.java [hunk 1]
```diff
@@ -28,5 +28,6 @@
 	Dummy, // not used anymore
 	Enum,
 	RestrictedString,
-	RestrictedText
+	RestrictedText,
+	Hidden
```
Comment (loomis) [inline jar-persistence/src/main/java/com/sixsq/slipstream/persistence/ParameterType.java]: @konstan Will this cause any migration issues at the database level?

Comment (konstan) [inline jar-persistence/src/main/java/com/sixsq/slipstream/persistence/ParameterType.java]: @loomis No, there will be no issues with this on DB level.

**Comment by konstan** [comment]
I'm compiling everything locally.  If everything compiles I can merge.  Or are you planning to push more into the branch?

**Comment by loomis** [comment]
You can merge it if it compiles correctly.  I'm resolving conflicts with this PR on the branch for another one.

**Pull request merged**